### PR TITLE
Fixes an error when trying to update a non-existing SecurityDocument

### PR DESCRIPTION
**Pull request by samniisan**
Fixes an error when trying to update a non-existing SecurityDocument (User|Profile|Role)

### CODE REVIEW

File: lib/api/controllers/securityController.js [hunk 1]
```diff
@@ -1122,9 +1116,14 @@ function formatUserSearchResult(kuzzle, formatter, raw) {
  * Sets the metadata for an update request
  * @param {object} securityDocument
  * @param {Request} request
+ * @param {string} context
```
Comment (dbengsch) [inline lib/api/controllers/securityController.js]: wording: `context` word should be kept for `userContext` only.

Comment (samniisan) [inline lib/api/controllers/securityController.js]: Updated.

**Comment by samniisan** [comment]
@scottinet Not sure I get your point, this PR still returns a NotFoundError, just moved things around

**Comment by ballinette** [comment]
indeed, the `NotFoundError` was only triggered for `updateRole` before this PR.
Now, the code is factorized and the error is triggered for both `updateUser` and `updateProfile` as well ;)

**Comment by scottinet** [comment]
Oh, right, I missed the updated code for `updateMetadata`. I thought your PR would allow updating non-existing documents by somehow creating them.

**Pull request merged**